### DISCUSSION TITLE
Issue with my S3-WIFI-ST datalogger.

### DISCUSSION THREAD

**Discussion by ccj2740** [Pho3niX90/solis_modbus#39] [General]
Thanks for this nice project. I am getting data from Solis Cloud with S3-WIFI-ST datalogger atm.
This data is delayed with 1-3min , so not useful for automation and charging my car.

I really want to get realtime dataflow , so this is why your solis_modbus is very interesting.

I have installed it already , but I get an error when I try connect to the IP ?

Here is some screenshot:
![64c4f0158033935afa7e1efdca5d45791c39e223](https://github.com/Pho3niX90/solis_modbus/assets/126421743/1381585c-7960-4ee5-af33-63ee892f7845)
![43b1e5aa0b80fa8ef0c378458d976019c8f252b2](https://github.com/Pho3niX90/solis_modbus/assets/126421743/f86bb67f-4e84-46df-a6b5-430f06188977)
![a68938d15185b6940b9b31c36a0173a4c8642541](https://github.com/Pho3niX90/solis_modbus/assets/126421743/ff016dec-9183-4e47-8a1c-815d2bb1498e)
![dfe39722d5d16b5a8d5da6ae5f0a63b679566e83_2_690x93](https://github.com/Pho3niX90/solis_modbus/assets/126421743/6c058e93-2da1-4fae-9d75-2249a91a0ae7)

**Comment by ccj2740** [comment]
Also I found this : https://github.com/hn/ginlong-solis and it seems like I need to use an ESPhome and soldering things. 
I am novice when it comes to modbus, so if that is required then I might just stay with my solis cloud.

Here is a picture of the datalogger manual:
![image](https://github.com/Pho3niX90/solis_modbus/assets/126421743/3a930a1e-e202-4212-92be-a7ff157ca1ca)

**Comment by 0rangutan** [comment]
The Solis Data Logger can only accept one connection at a time.  So using just the logger, you can use this integration OR Solis Cloud but not both.

If you want to use both simultaneously you will need to do a little DIY, eg. https://github.com/alienatedsec/solis-ha-modbus-cloud

**Comment by ccj2740** [comment]
> The Solis Data Logger can only accept one connection at a time. So using just the logger, you can use this integration OR Solis Cloud but not both.
> 
> If you want to use both simultaneously you will need to do a little DIY, eg. https://github.com/alienatedsec/solis-ha-modbus-cloud

Ahh ok , I need the Solis cloud because I have battery with time schedule for charge/discharge settings.
Guess I have to accept the 1-2min delay from Solis Cloud as I am not geek enough to do the DIY :(...

**Comment by 0rangutan** [comment]
Actually, with the latest release of this integration you can now configure charge and discharge times too.
So if there's nothing else you need Solis Cloud for, you can set this up as the only connection to your logger and enjoy!

**Comment by ccj2740** [comment]
> Actually, with the latest release of this integration you can now configure charge and discharge times too. So if there's nothing else you need Solis Cloud for, you can set this up as the only connection to your logger and enjoy!

Very interesting, but if I take it out of Solis cloud will I loose the possibility for firmware updates from Solis. Also I have some other configuration with SOC levels. However is it easy to just take it out and setup solis_modbus ? (sorry for my newbie questions)

**Comment by Pho3niX90** [comment]
> > Actually, with the latest release of this integration you can now configure charge and discharge times too. So if there's nothing else you need Solis Cloud for, you can set this up as the only connection to your logger and enjoy!
> 
> Very interesting, but if I take it out of Solis cloud will I loose the possibility for firmware updates from Solis. Also I have some other configuration with SOC levels. However is it easy to just take it out and setup solis_modbus ? (sorry for my newbie questions)

Firmware updates are still possible. I think this also depends on the specific support that is assisting in the firmware update. 

Example. I have requested 2 firmware updates in the last month. Once for the inverter, and on e for the datalogger. For the inverter, it was just pushed from their side without issues, for the datalogger, they requested it to be connected to their api.
For the latter, I just disabled the modbus, gave it 5mins to sync, and they issued the update.

**Comment by ccj2740** [comment]
@Pho3niX90 : Ahh then it make sense to switch my datalogger to LAN configuration instead. Is it easy to disconnect from Solis cloud and set it up to your solid_modbus ? Is there a guide for this ?

**Comment by kiloptero** [comment]
hii! maybe a stupid question.. you can use this integration and when only you need the firmware update disconnect the integration for a few minutes?

**Comment by Pho3niX90** [comment]
> hii! maybe a stupid question.. you can use this integration and when only you need the firmware update disconnect the integration for a few minutes?

Yes correct.

**Comment by kiloptero** [comment]
cant wait to install! i will share it
thanks a looot!!

**Comment by kiloptero** [comment]
another question.. ill have two dataloggers.. its possible to use 1 with solis cloud and other for ha.. just "swaping" when i needed? ill use 99% of the time with ha.. just solis for updates..

**Comment by Pho3niX90** [comment]
> another question.. ill have two dataloggers.. its possible to use 1 with solis cloud and other for ha.. just "swaping" when i needed? ill use 99% of the time with ha.. just solis for updates..

How will these data loggers be connected? 

So from the plugins side, it pulls the information from the datalogger, vs the datalogger pushing to it. So if one of your dataloggers has an open and unused modbus port, it should work fine.
